### PR TITLE
Jinghan/enrich entity for ListGroup and GetGroup

### PR DESCRIPTION
**Pull request by jinghancc**
This PR does:
- add argument `entityIDs` for db method `ListEntity`
- enrich entity for db method `ListGroup`, `GetGroup(ByName)`

related to #660

### CODE REVIEW

File: internal/database/metadata/postgres/entity.go [hunk 1]
```diff
@@ -73,11 +74,21 @@ func getEntityByName(ctx context.Context, sqlxCtx metadata.SqlxContext, name str
...
-func listEntity(ctx context.Context, sqlxCtx metadata.SqlxContext) (types.EntityList, error) {
-	query := "select * from entity"
+func listEntity(ctx context.Context, sqlxCtx metadata.SqlxContext, entityIDs *[]int) (types.EntityList, error) {
+	query := "SELECT * FROM entity"
+	var args []interface{}
+	var err error
+	if entityIDs != nil {
+		if len(*entityIDs) == 0 {
+			return nil, nil
+		}
```
Comment (lianxmfor) [inline internal/database/metadata/postgres/entity.go:84]: Maybe just checking `len(entityIDs) == 0` will work 🤔

Comment (jinghancc) [reply internal/database/metadata/postgres/entity.go:84]: it does not work, https://go.dev/play/p/tsALR2HcDmr

File: internal/database/metadata/store.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ type ReadStore interface {
 	CacheGetEntity(ctx context.Context, id int) (*types.Entity, error)
 	CacheGetEntityByName(ctx context.Context, name string) (*types.Entity, error)
 	CacheListEntity(ctx context.Context) types.EntityList
-	ListEntity(ctx context.Context) (types.EntityList, error)
+	ListEntity(ctx context.Context, entityIDs *[]int) (types.EntityList, error)
```
Comment (lianxmfor) [inline internal/database/metadata/store.go:24]: Maybe `ListEntity(ctx context.Context, entityIDs []int) (types.EntityList, error)` is better and there can be less checks in implement 🤔

Comment (jinghancc) [reply internal/database/metadata/store.go:24]: `entityIDs == nil` and `len(*entityIDs) == 0` represents two different conditions:
- `entityIDs == nil` means list all entities
- `len(*entityIDs) == 0`  means don't no entity

that's why we need a pointer here, see `ListFeatureOpt` for reference

Comment (lianxmfor) [reply internal/database/metadata/store.go:24]: well, understand. we need to know the user if used argument `entityIDs`

Comment (jinghancc) [reply internal/database/metadata/store.go:24]: yup, that's correct!

Comment (lianxmfor) [reply internal/database/metadata/store.go:24]: I feel that there are two things that are not so good about this way of writing:
1. pointer slices are not well constructed.
2. the method definition does not make it clear to the user how the `entityIDs` affect the result.

How do you feel about this?
```go
type ListEntityOpt struct {
      EntityIDs []int
      UseEntityIds bool
}
```

Comment (jinghancc) [reply internal/database/metadata/store.go:24]: hmm introducing another argument doesn't alleviate the complexity here. If we have more than one condition, we will double the arguments in `ListEntityOpt`

Comment (lianxmfor) [reply internal/database/metadata/store.go:24]: make sense

**Review by lianxmfor — APPROVED** [review]
lgtm

**Pull request merged**